### PR TITLE
Eslint for playground specific to typescript [do not merge]

### PR DESCRIPTION
**Pull request by digvijay-u**
## Changes
- Added the config file `playground.js` with similar config as `index`
- Divided the rules into specific files so that they can be reused in the future

### CODE REVIEW

File: playground.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "./rules/react-hooks.js",
+    "./rules/react.js",
+    "./rules/promise.js",
+    "./rules/import.js",
+    "./rules/prettier.js"
+  ],
+  plugins: [
+    "@typescript-eslint"
+  ],
+  env: {
```
Comment (digvijay-u) [inline playground.js]: These can be project specific, can be removed and added to the project's `.eslintrc`

File: index.js [hunk 2]
```diff
@@ -1,126 +1,19 @@
 module.exports = {
-  plugins: ['flowtype', 'import', 'prettier', 'promise', 'react', 'react-hooks'],
-  extends: ['plugin:flowtype/recommended', 'plugin:react/recommended', 'airbnb', 'prettier'],
+  plugins: [
```
Comment (yamov) [inline index.js]: You don't need that, do you?
```suggestion
  plugins: [
```

File: index.js [hunk 3]
```diff
@@ -1,126 +1,19 @@
...
+  ],
+  extends: [
+    'plugin:flowtype/recommended',
```
Comment (yamov) [inline index.js]: This should belong to the `rules/flow`, IMO

File: index.js [hunk 4]
```diff
@@ -1,126 +1,19 @@
 module.exports = {
-  plugins: ['flowtype', 'import', 'prettier', 'promise', 'react', 'react-hooks'],
-  extends: ['plugin:flowtype/recommended', 'plugin:react/recommended', 'airbnb', 'prettier'],
+  plugins: [
+  ],
+  extends: [
+    'plugin:flowtype/recommended', 
+    'plugin:react/recommended',
```
Comment (yamov) [inline index.js]: This should belong to the `rules/react`, IMO

File: index.js [hunk 5]
```diff
@@ -1,126 +1,19 @@
...
+    'plugin:react/recommended', 
+    'airbnb', 
+    'prettier',
+    './rules/prettier.js',
+    './rules/eslint.js',
+    './rules/import.js',
+    './rules/promise.js',
+    './rules/react.js',
+    './rules/react-hooks.js',
+    './rules/flow.js'
```
Comment (yamov) [inline index.js]: I'd stick to the name of the original plugin/extended config, to make it easier to follow.
```suggestion
    './rules/flowtype.js'
```

File: index.js [hunk 6]
```diff
@@ -1,126 +1,19 @@
 module.exports = {
-  plugins: ['flowtype', 'import', 'prettier', 'promise', 'react', 'react-hooks'],
-  extends: ['plugin:flowtype/recommended', 'plugin:react/recommended', 'airbnb', 'prettier'],
+  plugins: [
+  ],
+  extends: [
+    'plugin:flowtype/recommended', 
+    'plugin:react/recommended', 
+    'airbnb', 
+    'prettier',
```
Comment (yamov) [inline index.js]: This should belong to the `rules/prettier`, IMO

File: ts.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+module.exports = {
+  plugins: [
+    "@typescript-eslint"
+  ],
+  extends: [
+    "plugin:react/recommended",
```
Comment (yamov) [inline ts.js]: See comments from above.

File: ts.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  ],
+  rules: {
+  },
+  settings: {
+    react: {
+      pragma: "React",
+      version: "detect"
+    },
+    "import/resolver": {
+      node: {
```
Comment (yamov) [inline ts.js]: What is it needed for?

Comment (digvijay-u) [reply ts.js]: We need this to specify the extension of the files to be parsed by default it is `['js', 'jsx']`
https://github.com/benmosher/eslint-plugin-import#settings

File: index.js (hunk 6 above)
Comment (digvijay-u) [reply index.js]: For some reason extensions are not working if I move them to specific rule files, it seems the sequence is causing some rules to override (just a guess)

File: index.js (hunk 4 above)
Comment (digvijay-u) [reply index.js]: I agree but for some reason extensions are not working if I move them to specific rule files, it seems the sequence is causing some rules to override (just a guess)

File: index.js (hunk 2 above)
Comment (digvijay-u) [reply index.js]: removed 👍

File: playground.js (hunk 1 above)
Comment (digvijay-u) [reply playground.js]: moved 👍

File: index.js (hunk 6 above)
Comment (yamov) [reply index.js]: IIRC the sequence indeed matters in terms of which rule takes precedence, but don't fully follow why moving the `prettier` config to the getsafe prettier subconfig doesn't work.

**Comment by digvijay-u** [comment]
Tested with all other projects

image

File: ts.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    require.resolve('./rules/import.js'),
+    require.resolve('./rules/promise.js'),
+    require.resolve('./rules/react-hooks.js'),
+    require.resolve('./rules/prettier.js'),
+  ],
+  rules: {
+    'react/jsx-filename-extension': ['error', { extensions: ['.tsx'] }],
+    // Todo: investigate why eslint fails for typescript when
+    // turn on below prop
+    'import/no-unused-modules': ['off'],
```
Comment (yamov) [inline ts.js]: Nitpick: 
```suggestion
    'import/no-unused-modules': 'off',
```

**Pull request merged**